TFNetwork.py

```diff
@@ -383,18 +383,24 @@ def construct_layer(self, net_dict, name, get_layer=None, add_layer=None, check_
     if name in self._constructing_layers:
       raise NetworkConstructionDependencyLoopException(
         layer_name=name, constructing_layers=self._constructing_layers, net_dict=net_dict, network=self)
+    if not get_layer:
+      def get_layer(src_name):
+        return self.construct_layer(net_dict=net_dict, name=src_name)  # set get_layer to wrap construct_layer
     if name not in net_dict:
       if name == "data":
         layer_desc = {"class": "source", "from": []}
       elif name.startswith("data:"):
         layer_desc = {"class": "source", "data_key": name[len("data:"):], "from": []}
+      elif '/' in name:
+        # it may be a hierarchical path to a sub-layer, which should have been found by get_layer()
+        # but maybe it's not constructed yet, so try constructing the root layer
+        get_layer(name.split('/')[0])
+        # constructing the root layer should have constructed all its children
+        return get_layer(name)  # ...so try again now
```

Comment:
Do not call `get_layer(name)` here. This could lead to an infinite recursion (if this is not an existing hierarchical path).
Rather, assert that `name.split('/')[0] in self.layers` now, and then move on right here, so it will go into the code `if '/' in layer_name: ...` below.

Reply:
You're right. I ignored that case. 
See my newest change.